test/src/test/java/org/corfudb/infrastructure/logreplication/LogReplicationFSMTest.java

```diff
@@ -154,6 +159,44 @@ public void testLogReplicationFSMTransitions() throws Exception {
         transition(LogReplicationEventType.REPLICATION_STOP, LogReplicationStateType.INITIALIZED, true);
     }
 
+    /**
+     * Verify the lastSyncType flag is being initialized properly.
+     *
+     */
+    @Test
+    public void testFullSyncStatusOnInit() throws Exception {
+        initLogReplicationFSM(ReaderImplementation.EMPTY);
+
+        final Table<LogReplicationSession, LogReplicationMetadata.ReplicationStatus, Message> statusTable =
+                this.corfuStore.getTable(NAMESPACE, REPLICATION_STATUS_TABLE_NAME);
+
+        // Initial state: Initialized
+        LogReplicationState initState = fsm.getState();
+        assertThat(initState.getType()).isEqualTo(LogReplicationStateType.INITIALIZED);
+
+        transitionAvailable.acquire();
+
+        // Transition #1: Replication Stop (without any replication having started)
+        transition(LogReplicationEventType.REPLICATION_STOP, LogReplicationStateType.INITIALIZED);
+
+        // Default sync value is null so replication status should not be set at all.
+        final int expectedStatusTableLength = 0;
+        Assert.assertEquals(expectedStatusTableLength, statusTable.count());
+
+        // Transition #2: SNAPSHOT Sync Start
```

Comment:
We should also check that a transition to Log Entry Sync updates the status correctly.